### ISSUE TITLE
MSVC multi-process build logs parsed incorrectly

### ISSUE THREAD

**Issue by Mischa-Alff** [microsoft/vscode-cmake-tools#256]
Hi,

When building a project using the `Visual Studio 15 2017 Win64` generator, `vscode-cmake-tools` starts the build with the following command:

`cmake --build c:/users/foo/bar/build --target ALL_BUILD --config Debug -- /m /property:GenerateFullPaths=true`

This invokes `MSBuild.exe` with the `/m` option, which requests a multiprocess build, which in turn causes MSBuild to prefix all output with the build process it came from (simplified for demonstration purposes):

```
1>Project "C:\foo\bar\build\ALL_BUILD.vcxproj" on node 1 (default targets).
     1>Project "C:\foo\bar\build\ALL_BUILD.vcxproj" (1) is building "C:\foo\bar\build\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
     2>Done Building Project "C:\foo\bar\build\ZERO_CHECK.vcxproj" (default targets).
     1>Project "C:\foo\bar\build\ALL_BUILD.vcxproj" (1) is building "C:\foo\bar\build\bar.vcxproj" (4) on node 2 (default targets).
     4>Done Building Project "C:\foo\bar\build\bar.vcxproj" (default targets).
     1>Project "C:\foo\bar\build\ALL_BUILD.vcxproj" (1) is building "C:\foo\bar\build\example.vcxproj" (3) on node 3 (default targets).
     3>InitializeBuildStatus:
         Touching "example.dir\Debug\example.tlog\unsuccessfulbuild".
       CustomBuild:
         All outputs are up-to-date.
       ClCompile:
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.11.25503\bin\HostX86\x64\CL.exe /c /IC:\foo\bar\include /Zi /nologo /W3 /WX- /diagnostics:classic /Od /Ob0 /D WIN32 /D _WINDOWS /D "CMAKE_INTDIR=\"Debug\"" /D _MBCS /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /Fo"example.dir\Debug\\" /Fd"example.dir\Debug\vc141.pdb" /Gd /TP /errorReport:queue C:\foo\bar\examples\example.cpp
         example.cpp
     3>C:\foo\bar\include\bar.hpp(67): error C2429: language feature 'init-statements in if/switch' requires compiler flag '/std:c++latest' 
```

`vscode-cmake-tools` seems to parse the file path as including the `3>`, but later parses it correctly in MSBuild's error summary as such:

![capture](https://user-images.githubusercontent.com/1387364/32835399-3ea5333e-ca06-11e7-8a21-f53a6ace9a24.PNG)

**Comment by go2sh** [comment]
I have the same problem with 2017.

**Comment by vector-of-bool** [comment]
Thanks for the issue. I'll keep this ticket open until I add a unit test to check for this.

**Comment by Mischa-Alff** [comment]
Any news on this one? It still occurs for me.

**Comment by vector-of-bool** [comment]
Sorry, I haven't lifted the priority of this one for a while.

Diagnostics should receive a face-lift in 1.2.0, and I'll probably slurp this one into that milestone.

**Comment by Zingam** [comment]
Is this a related issue. I selected the command "CMake: Scan for Kits" on Linux:

[kit] Found Kit: GCC for c89 9.2.1
[kit] Found Kit: GCC for c99 9.2.1
[kit] Found Kit: GCC 9.2.1
[kit] Found Kit: GCC 9.2.1
[kit] Found Kit: GCC for x86_64-linux-gnu 9.2.1
[kit] Found Kit: GCC for x86_64-linux-gnu 9.2.1
[kit] Found Kit: GCC for c89 9.2.1
[kit] Found Kit: GCC for c99 9.2.1
[kit] Found Kit: GCC 9.2.1
[kit] Found Kit: GCC 9.2.1
[kit] Found Kit: GCC for x86_64-linux-gnu 9.2.1
[kit] Found Kit: GCC for x86_64-linux-gnu 9.2.1

**Comment by bobbrow** [comment]
@Zingam, no, this issue is about the problem matchers for MSVC builds.

**Comment by bobbrow** [comment]
This issue has been addressed in CMake Tools 1.2.0 which is available in the VS Code Marketplace or on the [Releases Page](https://github.com/microsoft/vscode-cmake-tools/releases/tag/v1.2.0)

Because the publisher id has changed, VS Code does not automatically upgrade you yet. The Marketplace team will enable this in the next few days.  If you want to install 1.2.0 now, please make sure you uninstall any earlier versions of CMake Tools manually to avoid any conflicts.
